sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/rest_client.hpp

```diff
@@ -1314,48 +1323,50 @@ namespace Azure { namespace Storage { namespace Blobs {
     private:
       std::string m_value;
     };
-    /**
-     * @brief An Azure Storage blob.
-     */
-    struct BlobItem final
-    {
-      /**
-       * Blob name.
-       */
-      std::string Name;
-      /**
-       * Indicates whether this blob was deleted.
-       */
-      bool IsDeleted = bool();
-      /**
-       * A string value that uniquely identifies a blob snapshot.
-       */
-      std::string Snapshot;
-      /**
-       * A string value that uniquely identifies a blob version.
-       */
-      Nullable<std::string> VersionId;
-      /**
-       * Indicates if this is the current version of the blob.
-       */
-      Nullable<bool> IsCurrentVersion;
-      /**
-       * Properties of a blob.
-       */
-      BlobItemDetails Details;
-      /**
-       * Indicates that this root blob has been deleted, but it has versions that are active.
-       */
-      Nullable<bool> HasVersionsOnly;
-      /**
-       * Size in bytes.
-       */
-      int64_t BlobSize = int64_t();
+    namespace _detail {
       /**
-       * Type of the blob.
+       * @brief An Azure Storage blob.
        */
-      Models::BlobType BlobType;
-    };
+      struct BlobItem final
+      {
+        /**
+         * Blob name.
+         */
+        BlobName Name;
+        /**
+         * Indicates whether this blob was deleted.
+         */
+        bool IsDeleted = bool();
+        /**
+         * A string value that uniquely identifies a blob snapshot.
+         */
+        std::string Snapshot;
+        /**
+         * A string value that uniquely identifies a blob version.
+         */
+        Nullable<std::string> VersionId;
+        /**
+         * Indicates if this is the current version of the blob.
+         */
+        Nullable<bool> IsCurrentVersion;
+        /**
+         * Properties of a blob.
+         */
+        BlobItemDetails Details;
+        /**
+         * Indicates that this root blob has been deleted, but it has versions that are active.
+         */
+        Nullable<bool> HasVersionsOnly;
+        /**
+         * Size in bytes.
+         */
+        int64_t BlobSize = int64_t();
```

Comment:
```suggestion
        int64_t BlobSize{};
```

You get the idea - I'm going to stop doing these at this point.